### PR TITLE
feat: IG android - open share screen

### PR DESCRIPTION
**Pull request by ddereszewski**
# Overview
This introduces a new feature to IG single share. 
I found out how to open share views on IG without providing media.  

# Test Plan
As added in docs

### CODE REVIEW

File: android/src/main/java/cl/json/social/InstagramShare.java [hunk 1]
```diff
@@ -24,7 +24,15 @@ public InstagramShare(ReactApplicationContext reactContext) {
...
     public void open(ReadableMap options) throws ActivityNotFoundException {
         super.open(options);
-        //  extra params here
+        try {
+            // passing instagram://share as url option opens Camera View for publishing
+            if (ShareIntent.hasValidKey("url", options) && options.getString("url").startsWith("instagram://")) {
+                this.getIntent().setAction(Intent.ACTION_VIEW);
+                this.getIntent().setData(Uri.parse(options.getString("url")));
+            }
+        }catch (Exception e){
```
Comment (mikehardy) [inline android/src/main/java/cl/json/social/InstagramShare.java:33]: Best to match existing spacing style

```suggestion
        } catch (Exception e) {
```

File: android/src/main/java/cl/json/social/InstagramShare.java [hunk 2]
```diff
@@ -24,7 +24,15 @@ public InstagramShare(ReactApplicationContext reactContext) {
...
+            // bad param passed
```
Comment (mikehardy) [inline android/src/main/java/cl/json/social/InstagramShare.java:34]: If these are simply consumed with no output whatsover, when the params required inevitably change upstream there will be no notice for anyone anywhere. So these should be logged somehow. The question is severity. For severity "error" means execution should halt as state is undefined - that is not correct here, "info" means no problem really just something you should know - that is also not quite right, "warning" means something unexpected and bad but we think we can continue. I think that fits.

```suggestion
            // bad param passed
            android.util.Log.warn("RNShare", e.getMessage());
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Interesting - thanks for posting this!

Seems good but spacing doesn't match project and I worry about just eating that exception... what do you think of my suggestions?

**Comment by ddereszewski** [comment]
pushed the fixed. 

I decided to allow `instagram://*` patter for Android as it works but it's nowhere documented. Folks might use it in some other way to achieve some other results. 

Also, I didn't want to introduce any other new param,  increase complexity and make params less generic (Story case). Therefore I suggest a solution to use URL

**Review by MateusAndrade — APPROVED** [review]
Nice! Thanks a lot @ddereszewski! 🚀

**Comment by MateusAndrade** [comment]
Do you think we can follow with this @mikehardy? 🤔

**Review by mikehardy — APPROVED** [review]
Sure seems fine by me - were suggestions were more sort of general and not related to the feature, I'll defer to anyone (which is certainly @ddereszewski here...) that's actually working on and testing it. I have no idea how Instagram works at an integration / API level haha

**Pull request merged**